workshops/gan/src/training.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torchvision
+from torch.utils.tensorboard import SummaryWriter  # TODO: use it
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from critics import FCCritic
+from generators import FCGenerator
+from dataset import FacesDataSet
+
+IMG_SIZE = 128
+CHANNELS = 3
+Z_SIZE = 100
+
+CUDA = True if torch.cuda.is_available() else False
+MAX_SUMMARY_IMAGES = 4
+LR = 1e-4
+EPOCHS = 20
+BATCH_SIZE = 64
+NUM_WORKERS = 8  # set to 0 when debugging
+CLIP_VALUE = 1e-2
+
+assert MAX_SUMMARY_IMAGES <= BATCH_SIZE
+
+def train():
+    summary_writer = SummaryWriter()
+
+    critic = FCCritic(IMG_SIZE, CHANNELS)
+    generator = FCGenerator(IMG_SIZE, CHANNELS, Z_SIZE)
+    if CUDA:
+        critic.cuda()
+        generator.cuda()
```

Comment:
Maybe to set up at the begining
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
and then
critic.to(device)
generator.to(device)
etc.